cppcache/src/CacheFactory.cpp

```diff
@@ -109,39 +132,6 @@ Cache CacheFactory::create() const {
   return cache;
 }
 
-Cache CacheFactory::create(
-    std::string name,
-    const std::shared_ptr<CacheAttributes>& attrs /*= nullptr*/) const {
-  auto cache = Cache(std::move(name), dsProp, ignorePdxUnreadFields,
-                     pdxReadSerialized, authInitialize);
-  cache.m_cacheImpl->setAttributes(attrs);
-
-  try {
-    auto&& cacheXml =
-        cache.getDistributedSystem().getSystemProperties().cacheXMLFile();
-    if (!cacheXml.empty()) {
-      cache.initializeDeclarativeCache(cacheXml);
-    } else {
-      cache.m_cacheImpl->initServices();
-    }
-  } catch (const apache::geode::client::RegionExistsException&) {
-    LOGWARN("Attempt to create existing regions declaratively");
-  } catch (const apache::geode::client::Exception&) {
-    if (!cache.isClosed()) {
-      cache.close();
-    }
-    throw;
-  } catch (...) {
-    if (!cache.isClosed()) {
-      cache.close();
-    }
-    throw apache::geode::client::UnknownException(
-        "Exception thrown in CacheFactory::create");
-  }
-
-  return cache;
-}
-
 std::shared_ptr<CacheFactory> CacheFactory::set(const std::string& name,
```

Comment:
If Cache name is linked to DS name, doesn't this need to flow through to the DS...?

Reply:
Yes, but I am not sure what you are referring to here?

Reply:
CacheFactory::set() takes a name parameter, which is inserted into the dsProperties... is that guaranteed to update the name the DS uses?

Reply:
This method takes a name/value pair to update ds properties. It is not explicitly setting the "ds name". 

The fact that this is the worst way to handle configuration is outside the scope of this issue.

Reply:
fair enough...